File: 217.contains-duplicate.java
/*
 * @lc app=leetcode id=217 lang=java
 *
 * [217] Contains Duplicate
 */
class Solution {
    public boolean containsDuplicate(int[] nums) {
        if (nums == null) return false;
        Set<Integer> hist = new HashSet<>();
        for (int item : nums) {
            if (hist.contains(item)) return true;
            hist.add(item);
        }
        return false;
    }
}

File: go/217.contains-duplicate.go
/*
 * @lc app=leetcode id=217 lang=golang
 *
 * [217] Contains Duplicate
 */

// @lc code=start
func containsDuplicate(nums []int) bool {
	set := make(map[int]struct{})
	for i := 0; i < len(nums); i++ {
		if _, b := set[nums[i]]; b {
			return true
		}
		set[nums[i]] = struct{}{}
	}
	return false
}

// @lc code=end

